connector/sumconnector/config.go

```diff
@@ -24,3 +34,88 @@ type AttributeConfig struct {
+			return fmt.Errorf("spans condition: metric %q: %w", name, err)
+		}
+		if err := info.validateAttributes(); err != nil {
+			return fmt.Errorf("spans attributes: metric %q: %w", name, err)
+		}
+		if info.SourceAttribute == "" {
+			return fmt.Errorf("spans: metric source_attribute missing")
+		}
+	}
+	for name, info := range c.SpanEvents {
+		if name == "" {
+			return fmt.Errorf("spanevents: metric name missing")
+		}
+		if _, err := filterottl.NewBoolExprForSpanEvent(info.Conditions, filterottl.StandardSpanEventFuncs(), ottl.PropagateError, component.TelemetrySettings{Logger: zap.NewNop()}); err != nil {
+			return fmt.Errorf("spanevents condition: metric %q: %w", name, err)
+		}
+		if err := info.validateAttributes(); err != nil {
+			return fmt.Errorf("spanevents attributes: metric %q: %w", name, err)
+		}
+		if info.SourceAttribute == "" {
+			return fmt.Errorf("spanevents: metric source_attribute missing")
+		}
+	}
+	for name, info := range c.Metrics {
+		if name == "" {
+			return fmt.Errorf("metrics: metric name missing")
+		}
+		if _, err := filterottl.NewBoolExprForMetric(info.Conditions, filterottl.StandardMetricFuncs(), ottl.PropagateError, component.TelemetrySettings{Logger: zap.NewNop()}); err != nil {
+			return fmt.Errorf("metrics condition: metric %q: %w", name, err)
+		}
+		if len(info.Attributes) > 0 {
+			return fmt.Errorf("metrics attributes not supported: metric %q", name)
+		}
+		if info.SourceAttribute == "" {
+			return fmt.Errorf("metrics: metric source_attribute missing")
+		}
+	}
+	for name, info := range c.DataPoints {
+		if name == "" {
+			return fmt.Errorf("datapoints: metric name missing")
+		}
+		if _, err := filterottl.NewBoolExprForDataPoint(info.Conditions, filterottl.StandardDataPointFuncs(), ottl.PropagateError, component.TelemetrySettings{Logger: zap.NewNop()}); err != nil {
+			return fmt.Errorf("datapoints condition: metric %q: %w", name, err)
+		}
+		if err := info.validateAttributes(); err != nil {
+			return fmt.Errorf("spans attributes: metric %q: %w", name, err)
+		}
+		if info.SourceAttribute == "" {
+			return fmt.Errorf("datapoints: metric source_attribute missing")
+		}
+	}
+	for name, info := range c.Logs {
+		if name == "" {
+			return fmt.Errorf("logs: metric name missing")
+		}
+		if _, err := filterottl.NewBoolExprForLog(info.Conditions, filterottl.StandardLogFuncs(), ottl.PropagateError, component.TelemetrySettings{Logger: zap.NewNop()}); err != nil {
+			return fmt.Errorf("logs condition: metric %q: %w", name, err)
+		}
+		if err := info.validateAttributes(); err != nil {
+			return fmt.Errorf("logs attributes: metric %q: %w", name, err)
+		}
+		if info.SourceAttribute == "" {
+			return fmt.Errorf("logs: metric source_attribute missing")
+		}
+	}
+	return nil
+}
+
+func (i *MetricInfo) validateAttributes() error {
+	for _, attr := range i.Attributes {
+		if attr.Key == "" {
+			return fmt.Errorf("attribute key missing")
```

Comment:
It would be good to make it clear that `key` here is referring to the config option under `attributes`.

Reply:
changed to AttrKey`

Reply:
Oh, sorry, I was meaning the error message here, not the variable name. I think `attr.Key` makes sense, and probably prefer it over `attr.Attrkey`, to be honest.

It may be alright to just revert the name change and leave the error message as-is. If it's ever unclear users can file an enhancement request.

Reply:
Apologies for missing this reply and misinterpreting! Reverted the `Attrkey` symbol change.

I was trying to think of a better wording for this error but not sure how to get the message across without being really verbose? I'll leave it as is for now and yeah if someone needs clarification I'll happily try to find a better way to word the error.

Thanks again!